sql/core/src/test/scala/org/apache/spark/sql/execution/command/v1/ShowPartitionsSuite.scala

```diff
@@ -93,4 +93,26 @@ class ShowPartitionsSuite extends ShowPartitionsSuiteBase with CommandSuiteBase
       assert(sql("SHOW PARTITIONS part_datasrc").count() == 3)
     }
   }
+
+  test("null and empty string as partition values") {
+    import testImplicits._
+    withTable("t") {
+      val df = Seq(
+        (0, ""),
+        (1, null)
+      ).toDF("a", "part")
```

Comment:
In this case, shall we use a one-liner because it's simpler?
```scala
- val df = Seq(
-   (0, ""),
-   (1, null)
- ).toDF("a", "part")
+ val df = Seq((0, ""), (1, null)).toDF("a", "part")
```

Reply:
+1